DigitalLearningSolutions.Data/Services/DelegateApprovalsService.cs

```diff
@@ -30,5 +44,69 @@ public List<(DelegateUser delegateUser, List<CustomPromptWithAnswer> prompts)> G
 
             return usersWithPrompts;
         }
+
+        public void ApproveDelegate(int delegateId)
+        {
+            var delegateUser = userDataService.GetDelegateUserById(delegateId);
+
+            if (delegateUser == null)
+            {
+                throw new UserAccountNotFoundException($"Delegate user id {delegateId} not found");
+            }
+            if (delegateUser.Approved)
+            {
+                logger.LogWarning($"Delegate user id {delegateId} already approved.");
+            }
+            else
+            {
+                userDataService.ApproveDelegateUsers(new[] { delegateId });
+                SendApprovalEmail(delegateUser);
+            }
+        }
+
+        public void ApproveAllUnapprovedDelegatesForCentre(int centreId)
+        {
+            var delegateUsers = userDataService.GetUnapprovedDelegateUsersByCentreId(centreId);
+
+            userDataService.ApproveDelegateUsers(delegateUsers.Select(d => d.Id));
+            foreach (var delegateUser in delegateUsers)
+            {
+                SendApprovalEmail(delegateUser);
+            }
+        }
+
+        private void SendApprovalEmail(DelegateUser delegateUser)
+        {
+            if (string.IsNullOrWhiteSpace(delegateUser.EmailAddress))
+            {
+                logger.LogWarning($"Delegate user id {delegateUser.Id} has no email associated with their account.");
+            }
+            else
+            {
+                var delegateApprovalEmail = GenerateDelegateApprovalEmail(delegateUser.CandidateNumber, delegateUser.EmailAddress);
+                emailService.SendEmail(delegateApprovalEmail);
+            }
+        }
+
+        private static Email GenerateDelegateApprovalEmail(
+            string candidateNumber,
+            string emailAddress)
+        {
+            string emailSubject = "Digital Learning Solutions Registration Approved";
+
+            var body = new BodyBuilder
+                {
+                    TextBody = $@"Your Digital Learning Solutions registration has been approved by your centre administrator.
+                            You can now login to the Digital Learning Solutions learning materials using your e-mail address or your Delegate ID number <b>""{candidateNumber}""</b> and the password you chose during registration.
+                            For more assistance in accessing the materials, please contact your Digital Learning Solutions centre.",

```

Comment:
Might be worth adding a link to the sign in page here - could you send a screenshot of the email to the HEE external Slack and ask if it's something they'd like? :)